### DISCUSSION TITLE
ECONNREFUSED connecting to local PostgreSQL from Docker

### DISCUSSION THREAD

**Discussion by ultranol** [Budibase/budibase#9345] [Help] [answered]
Trying to add a datasource to my first app and unable to connect to my local PostgreSQL. I am aware that "localhost" on the "host" field refers to the Docker container itself. However I tried everything I could find: host.docker.internal gives me ENOTFOUND, and I found https://github.com/Budibase/budibase/discussions/7651 saying in Linux (I am on Pop_OS/Ubuntu 22.04) I should use 172.17.0.1, but I get ECONNREFUSED using that, as if I was trying localhost. Any other ideas?

**Comment by ultranol** [comment, marked as answer]
Solved myself - had to open PostgreSQL to accept connections from other places in the network. I had assumed because I was able to connect to it through my local DB management program, PGSQL was open; however, Docker is like a different machine trying to access it.

**Reply by smokey58au** [reply]
You are a legend!  I have spent 2 days running down every rabbit hole concerning network hosts, extra-hosts etc etc, and kept hitting the connection refused error.  I'm running Budibase in a docker container inside a Ubuntu virtual machine, and my database server (MariaDB) is running in the VM itself, so I knew the issue was somewhere in the localhost connection/configuration, but it turns out I kept missing this key step above in MariaDB.  I had to open MariaDB to outside connections and then create a db user with access from the virtual machine, and all was good again.  If anyone else has this problem, here's the link to the key information that @ultranol 's solution pointed me to.

https://mariadb.com/kb/en/configuring-mariadb-for-remote-client-access/
